algoliasearch-common/src/main/java/com/algolia/search/inputs/query_rules/QueryRule.java

```diff
@@ -0,0 +1,166 @@
+package com.algolia.search.inputs.query_rules;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.List;
+
+public class QueryRule {
```

Comment:
I wonder if this class, which closely mirrors the JSON schema, provides the best developer experience. I chose `if` and `then` in the schema because it makes rules data relatively straightforward to read. But when it comes to code... maybe it's not the best idea I ever had. 😆 => I would go for `Condition` and `Consequence`.

IMHO, it would also be OK to use public fields for the sub-objects, especially if they are declared `final`, as they are mandatory anyway.

So, instead of `rule.getIfQueryRule().setPattern("foo bar")` you could write `rule.condition.setPattern("foo bar")`.

This doesn't apply to the description as it's optional and atomic (i.e. not an aggregate).